mpathconfig/configresource.go

```diff
@@ -237,15 +237,18 @@ func addOption(section *Section, option string) {
 	}
 }
 
+//checks for the section
 func isSection(line string) bool {
-	line = strings.TrimSpace(line)
+	line = strings.TrimSpace(line)	
 	prefixes := []string{"defaults", "blacklist", "blacklist_exceptions", "devices", "device", "multipaths", "multipath"}
-	for _, prefix := range prefixes {
-		if strings.HasPrefix(line, prefix) {
+        for _, prefix := range prefixes {
+		r := regexp.MustCompile("^"+prefix+"\\s*{")
```

Comment:
Mustcompile could panic , rather use Compile with appropriate error handling

Reply:
We've tried to move to Compile from MustCompile for same reason for safety.

 If you want to use the same could you please move it to the const/var section so that it is compiled early? You can refer to linux/multipath.go or linux/device.go.

Reply:
would this be fine?
r, err := regexp.Compile(""^"+prefix+"\\s*{"")
if err != nil {
    log.Fatal(err)
}

Reply:
Sorry..I won't be using the log.Fatal(err)

Reply:
Yes.